src/client.rs

```diff
@@ -194,3 +221,34 @@ async fn handle_response_test<T: Send + for<'de> Deserialize<'de> + 'static>(
         }
     }
 }
+
+struct ProgressReader {
+    inner: Vec<u8>,
+    progress_bar: ProgressBar,
+    bytes_read: u64,
+    position: usize,
+    chunk_size: usize,
+}
+
+impl AsyncRead for ProgressReader {
+    fn poll_read(
+        mut self: Pin<&mut Self>,
+        _cx: &mut Context<'_>,
+        buf: &mut tokio::io::ReadBuf<'_>,
+    ) -> Poll<tokio::io::Result<()>> {
+        let start = buf.filled().len();
+
+        let remaining = self.inner.len() - self.position;
+        let to_read = self.chunk_size.min(remaining.min(buf.remaining()));
+        let end = self.position + to_read;
+        buf.put_slice(&self.inner[self.position..end]);
+        self.position = end;
+
+        let end = buf.filled().len();
+        let new_bytes = (end - start) as u64;
+        self.bytes_read += new_bytes;
```

Comment:
`new_bytes` should be the same as `to_read` so the first two lines can be removed.
`self.bytes_read` is the same as position so we should also be able to remove it.

Reply:
Resolved in [296bed4](https://github.com/ChihChengLiang/karma_calculator/pull/23/commits/296bed4f9230ab5aca3d3cc5065463e38d458365) too